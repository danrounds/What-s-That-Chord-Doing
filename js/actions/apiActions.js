import { getReqUserScores, getReqHighScores, postReqAccount,
         putReqUserScores, putReqAccountPassword, deleteReqAccount }
from '../apiCalls';

const getScores = {
    getUserScores: (name, password) =>
        dispatch => {
            dispatch(getScores.getUserScoresPending());
            return getReqUserScores(name, password)
                .then(scores => dispatch(getScores.getUserScoresSuccess(scores)))
                .catch((e) => dispatch(getScores.getUserScoresFailure(e)));
        },

    GET_USER_SCORES_PENDING: 'GET_USER_SCORES_PENDING',
    getUserScoresPending: () => ({
        type: GET_USER_SCORES_PENDING
    }),

    GET_USER_SCORES_SUCCESS: 'GET_USER_SCORES_SUCCESS',
    getUserScoresSuccess: (scores) => ({
        type: GET_USER_SCORES_SUCCESS,
        scores
    }),

    GET_USER_SCORES_FAILURE: 'GET_USER_SCORES_FAILURE',
    getUserScoresFailure: (error) => ({
        type: GET_USER_SCORES_FAILURE,
        error
    }),
};

export const LOG_OFF = 'LOG_OFF';
export const logOff = () => ({
    type: LOG_OFF
});

const updateScores = {
    updateUserScores: (name, password, scores) =>
        dispatch => putReqUserScores(name, password, scores)
        .then(() => dispatch(updateScores.updateUserScoresSuccess(scores)))
        .catch((e) => dispatch(updateScores.updateUserScoresFailure(e))),

    UPDATE_USER_SCORES_SUCCESS: 'UPDATE_USER_SCORES_SUCCESS',
    updateUserScoresSuccess: () => ({
        type: UPDATE_USER_SCORES_SUCCESS
    }),

    UPDATE_USER_SCORES_FAILURE: 'UPDATE_USER_SCORES_FAILURE',
    updateUserScoresFailure: (error) => ({
        type: UPDATE_USER_SCORES_FAILURE,
        error
    }),
};

const getHighScores_ = {
    getHighScores: (gameType) =>
        dispatch => {
            dispatch(getHighScores_.getHighScoresPending());
            return getReqHighScores(gameType)
                .then(highScores => dispatch(getHighScores_.getHighScoresSuccess(highScores)))
                .catch((e) => dispatch(getHighScores_.getHighScoresFailure(e)));
        },

    GET_HIGH_SCORES_PENDING: 'GET_HIGH_SCORES_PENDING',
    getHighScoresPending: () => ({
        type: GET_HIGH_SCORES_PENDING
    }),

    GET_HIGH_SCORES_SUCCESS: 'GET_HIGH_SCORES_SUCCESS',
    getHighScoresSuccess: (highScores) => ({
        type: GET_HIGH_SCORES_SUCCESS,
        highScores
    }),

    GET_HIGH_SCORES_FAILURE: 'GET_HIGH_SCORES_FAILURE',
    getHighScoresFailure: (error) => ({
        type: GET_HIGH_SCORES_FAILURE,
        error
    }),
}

const makeAccount = {
    makeUserAccount: (name, password) =>
        dispatch => postReqAccount(name, password)
        .then(() => dispatch(makeAccount.makeUserAccountSuccess()))
        .catch(() => dispatch(makeAccount.makeUserAccountFailure())),

    MAKE_USER_ACCOUNT_SUCCESS: 'MAKE_USER_ACCOUNT_SUCCESS',
    makeUserAccountSuccess: () => ({
        type: MAKE_USER_ACCOUNT_SUCCESS
    }),

    MAKE_USER_ACCOUNT_FAILURE: 'MAKE_USER_ACCOUNT_FAILURE',
    makeUserAccountFailure: (error) => ({
        type: MAKE_USER_ACCOUNT_FAILURE,
        error
    }),
};

const changePass = {
    changeUserPassword: (name, password) =>
        dispatch => putReqAccountPassword(name, password)
        .then(() => dispatch(changePass.changeUserPasswordSuccess()))
        .catch((e) => dispatch(changePass.changeUserPasswordFailure())),

    CHANGE_USER_PASSWORD_SUCCESS: 'CHANGE_USER_PASSWORD_SUCCESS',
    changeUserPasswordSuccess: () => ({
        type: CHANGE_USER_PASSWORD_SUCCESS
    }),

    CHANGE_USER_PASSWORD_FAILURE: 'CHANGE_USER_PASSWORD_FAILURE',
    changeUserPasswordFailure: (error) => ({
        type: CHANGE_USER_PASSWORD_FAILURE,
        error
    }),
};

///
///
// I've not bothered to write an action for account deletion, because that
// doesn't belong in the MVP client

// We end up with a bit more code by grouping associated actions together, but
// it at least tells us which goes with which
export const { getUserScores, GET_USER_SCORES_PENDING, getUserScoresPending,
               GET_USER_SCORES_SUCCESS, getUserScoresSuccess,
               GET_USER_SCORES_FAILURE, getUserScoresFailure } = getScores;

export const { updateUserScores, UPDATE_USER_SCORES_SUCCESS,
               updateUserScoresSuccess, UPDATE_USER_SCORES_FAILURE,
               updateUserScoresFailure } = updateScores;

export const { getHighScores, GET_HIGH_SCORES_PENDING, getHighScoresPending,
               GET_HIGH_SCORES_SUCCESS, getHighScoresSuccess,
               GET_HIGH_SCORES_FAILURE, getHighScoresFailure } = getHighScores_;

export const { makeUserAccount, MAKE_USER_ACCOUNT_SUCCESS,
               makeUserAccountSuccess, MAKE_USER_ACCOUNT_FAILURE,
               makeUserAccountFailure } = makeAccount;

export const { changeUserPassword, CHANGE_USER_PASSWORD_SUCCESS,
               changeUserPasswordSuccess, CHANGE_USER_PASSWORD_FAILURE,
               changeUserPasswordFailure } = changePass;
