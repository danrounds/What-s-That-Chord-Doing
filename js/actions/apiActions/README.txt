NOTES:
=========================================================================
The logInSuccess, logOff, makeUserAccount actions success all update
`authToken` and `name` fields in localStorage. These are retrieved as our
initial state for our reducer.

I've not bothered to write an action for account deletion, because that
doesn't belong in the MVP client. If it ever gets written, in belongs in
./actionsAccountAndPassword.js

-DR, 2019
