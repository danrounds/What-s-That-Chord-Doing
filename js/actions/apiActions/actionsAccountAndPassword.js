import { httpPostReqAccount, httpPutReqAccountPassword, httpDeleteReqAccount } from '../../apiCalls';
import * as types from './types';

export function makeUserAccount(name, password) {
    return dispatch => {
        dispatch(makeUserAccountPending());
        return httpPostReqAccount(name, password)
            .then(token => dispatch(makeUserAccountSuccess(name, token)))
            .catch((e) => dispatch(makeUserAccountFailure(e)));
    };
}

export const makeUserAccountPending = () => ({
    type: types.MAKE_USER_ACCOUNT_PENDING,
});

export const makeUserAccountSuccess = (name, token) => {
    localStorage.setItem('@WTCD/authToken', token);
    localStorage.setItem('@WTCD/name', name);
    return {
        type: types.MAKE_USER_ACCOUNT_SUCCESS,
        name,
        token,
    };
};

export const makeUserAccountFailure = (error) => ({
    type: types.MAKE_USER_ACCOUNT_FAILURE,
    error,
});

////////////////////
// This hasn't actually been incorporated into the client \/
export function changeUserPassword(name, password) {
    return dispatch => httpPutReqAccountPassword(name, password)
        .then(() => dispatch(changeUserPasswordSuccess()))
        .catch((e) => dispatch(changeUserPasswordFailure()));
}

export const changeUserPasswordSuccess = () => ({
    type: types.CHANGE_USER_PASSWORD_SUCCESS,
});

export const changeUserPasswordFailure = (error) => ({
    type: types.CHANGE_USER_PASSWORD_FAILURE,
    error,
});
