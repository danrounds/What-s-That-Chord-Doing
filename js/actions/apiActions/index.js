export * from './actionsAccountAndPassword';
export * from './actionsLogInOff';
export * from './actionsScores';
