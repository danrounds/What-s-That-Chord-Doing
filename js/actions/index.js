import chordGetter from '../music-logic/';

export const GET_KEY_PRESS = 'GET_KEY_PRESS';
export const getKeyPress = (keyValue) => ({
    type: GET_KEY_PRESS,
    keyValue
});

export const TOGGLE_KEYBOARD_SHORTCUT_DISPLAY = 'TOGGLE_KEYBOARD_SHORTCUT_DISPLAY';
export const toggleKeyboardShortcutDisplay = (displayBool) => ({
    type: TOGGLE_KEYBOARD_SHORTCUT_DISPLAY,
    displayBool
});


export const START_NEW_GAME = 'START_NEW_GAME';
export const startNewGame = (gameType, inversions) => {
    let {keyNameReadable, keyNameNotation, introChordSequence, chordSubset} = chordGetter.init(gameType, inversions);
    let {chordName, currentChordNumeral, bassNote, trebleNotes, accidentals} = chordGetter.getChord();
    return {
        type: START_NEW_GAME,
        inversions,
        gameType,
        keyNameReadable,
        keyNameNotation,
        introChordSequence,
        chordSubset,
        chordName,
        currentChordNumeral,
        bassNote,
        trebleNotes,
        accidentals
    };
};

export const GET_NEXT_QUESTION = 'GET_NEXT_QUESTION';
export const getNextQuestion = () => {
    let {chordName, currentChordNumeral, bassNote, trebleNotes, accidentals} = chordGetter.getChord();
    return {
        type: GET_NEXT_QUESTION,
        chordName,
        currentChordNumeral,
        bassNote,
        trebleNotes,
        accidentals
    };
};

export const INCREMENT_GUESS_N = 'INCREMENT_GUESS_N';
export const incrementGuessN = () => ({
    type: INCREMENT_GUESS_N
});

export const MARK_TURN_CORRECT = 'MARK_TURN_CORRECT';
export const markTurnCorrect = () => ({
    type: MARK_TURN_CORRECT
});
