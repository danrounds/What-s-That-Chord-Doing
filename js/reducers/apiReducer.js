import * as actions from '../actions/apiActions';

const initialState = {
    highScores: [],
    userScores: {},
    error: null,
    pending: null,
};

function copyUserScores(obj) {
    if (!obj.name) { return {}; }

    const newObj = {name: obj.name}, scores = {};
    const keys = Object.keys(obj.scores);
    for (let i in keys) {
        scores[keys[i]] = Object.assign({}, scores[keys[i]], obj.scores[keys[i]]);
    }
    newObj.scores = scores;
    return newObj;
}

function copyHighScores(oldScores) {
    return oldScores.map(score => copyUserScores(score));
}

// Our reducer for the API interaction part of our app
export default (state=initialState, action) => {
    switch(action.type) {

    case 'GET_USER_SCORES_PENDING':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: copyUserScores(state.userScores),
            error: false,
            pending: true,
        };

    case 'GET_USER_SCORES_SUCCESS':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: action.scores,
            error: false,
            pending: false,
        };

    case 'GET_USER_SCORES_FAILURE':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: copyUserScores(state.userScores),
            error: action.error,
            pending: false,
        };

        //////////////////////////////////////////////
        //////////////////////////////////////////////

    case 'UPDATE_USER_SCORES_SUCCESS':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: copyUserScores(state.userScores),
            error: false,
            pending: false,
        };

    case 'UPDATE_USER_SCORES_FAILURE':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: copyUserScores(state.userScores),
            error: action.error,
            pending: false,
        };

        //////////////////////////////////////////////
        //////////////////////////////////////////////

    case 'GET_HIGH_SCORES_PENDING':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: copyUserScores(state.userScores),
            error: false,
            pending: true,
        };

    case 'GET_HIGH_SCORES_SUCCESS':
        return {
            highScores: action.highScores,
            userScores: copyUserScores(state.userScores),
            error: false,
            pending: false,
        };

    case 'GET_HIGH_SCORES_FAILURE':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: copyUserScores(state.userScores),
            error: action.error,
            pending: false,
        };

        //////////////////////////////////////////////
        //////////////////////////////////////////////

    case 'MAKE_USER_ACCOUNT_SUCCESS':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: copyUserScores(state.userScores),
            error: false,
            pending: false,
        };

    case 'MAKE_USER_ACCOUNT_FAILURE':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: copyUserScores(state.userScores),
            error: action.error,
            pending: false,
        };

        //////////////////////////////////////////////
        //////////////////////////////////////////////
        
    case 'CHANGE_USER_PASSWORD_SUCCESS':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: copyUserScores(state.userScores),
            error: false,
            pending: false,
        };

    case 'CHANGE_USER_PASSWORD_FAILURE':
        return {
            highScores: copyHighScores(state.highScores),
            userScores: copyUserScores(state.userScores),
            error: true,
            pending: false,
        };

    default:
        return state;
    }
};
