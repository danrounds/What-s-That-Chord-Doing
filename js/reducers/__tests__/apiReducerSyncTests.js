import apiReducer from '../apiReducer';
import * as actions from '../../actions/apiActions';

const initialState = {
    authToken: null,
    name: 'null',
    highScores: [],
    lastHighScoreAccessed: null,
    myScores: {},
    error: null,
    pending: null,
};

const myScores = {
    'name': 'joey',
    'scores': {
        'allChordsInv': {
            'totalClicks': 33,
            'nAnsweredRight': 13,
            'nQuestionNumber': 20,
            'winRatio': 0.3
        },
        'easyMajor': {
            'totalClicks': 22,
            'nAnsweredRight': 14,
            'nQuestionNumber': 20,
            'winRatio': 0.4
        }
    }
};

const highScores = [
    {
        'name': 'seofij',
        'scores': {
            'intermediateMinor': {
                'totalClicks': 4127,
                'nAnsweredRight': 210,
                'nQuestionNumber': 1000,
                'winRatio': 1
            }
        }
    },
    {
        'name': 'jake',
        'scores': {
            'intermediateMinor': {
                'totalClicks': 123466,
                'nAnsweredRight': 1245,
                'nQuestionNumber': 1291,
                'winRatio': 0.9612403100775194
            }
        }
    },
    {
        'name': 'aseos25DSDfj_',
        'scores': {
            'intermediateMinor': {
                'totalClicks': 433,
                'nAnsweredRight': 90,
                'nQuestionNumber': 100,
                'winRatio': 0.9
            }
        }
    }
];

const stateWithScores = {
    name: 'joey',
    lastHighScoreAccessed: 'hardMajor',
    pending: false,
    error: false,
    myScores: {
        'name': 'joey',
        'scores': {
            'allChordsInv': {
                'totalClicks': 22,
                'nAnsweredRight': 3,
                'nQuestionNumber': 10,
                'winRatio': 0.3
            },
            'easyMajor': {
                'totalClicks': 12,
                'nAnsweredRight': 4,
                'nQuestionNumber': 10,
                'winRatio': 0.4
            }
        }
    },

    highScores: [
        {
            'name': 'ijijijij',
            'scores': {
                'intermediateMinor': {
                    'totalClicks': 47,
                    'nAnsweredRight': 10,
                    'nQuestionNumber': 10,
                    'winRatio': 1
                }
            }
        },
        {
            'name': 'bobby4',
            'scores': {
                'intermediateMinor': {
                    'totalClicks': 12346,
                    'nAnsweredRight': 124,
                    'nQuestionNumber': 129,
                    'winRatio': 0.9612403100775194
                }
            }
        },
        {
            'name': 'aseofj_',
            'scores': {
                'intermediateMinor': {
                    'totalClicks': 43,
                    'nAnsweredRight': 9,
                    'nQuestionNumber': 10,
                    'winRatio': 0.9
                }
            }
        }
    ]
};

describe('apiReducer -- the syncronous bits', () => {
    let state;
    beforeEach(() => state = apiReducer(initialState, {}));

    it('should have a consistent initial state', () => {
        expect(state).toMatchSnapshot();
    });

    // log-in/log-off actions:
    it('should pend while we wait for log-in status', () => {
        state = apiReducer(initialState, actions.logInPending());
        expect(state).toMatchSnapshot();
    });

    it('should tell us when login-in succeeds', () => {
        state = apiReducer(initialState, actions.logInSuccess('jerry', 'some_string_of_token_or_other2DFS3'));
        expect(state).toMatchSnapshot();
    });

    it('should tell us when login-in fails', () => {
        state = apiReducer(initialState, actions.logInFailure(401));
        expect(state).toMatchSnapshot();
    });

    it('should log off', () => {
        state = apiReducer(stateWithScores, actions.logOff());
        expect(state).toMatchSnapshot();
    });


    // GET my-scores actions:
    it('should pend while we wait to GET my-scores', () => {
        state = apiReducer(stateWithScores, actions.getMyScoresPending());
        expect(state).toMatchSnapshot();
    });

    it('should tell us when we successfully GET my-scores', () => {
        state = apiReducer(stateWithScores, actions.getMyScoresSuccess(myScores));
        expect(state).toMatchSnapshot();
    });

    it('should tell us when we fail to GET my-scores', () => {
        state = apiReducer(stateWithScores, actions.getMyScoresFailure('here\'s an error'));
        expect(state).toMatchSnapshot();
    });


    // PUT my-scores actions:
    it('should tell us when we PUT our updated scores', () => {
        state = apiReducer(stateWithScores, actions.updateMyScoresSuccess());
        expect(state).toMatchSnapshot();
    });

    it('should tell us when we fail to PUT our updated scores', () => {
        state = apiReducer(stateWithScores, actions.updateMyScoresFailure(400));
        expect(state).toMatchSnapshot();
    });


    // GET high-scores actions:
    it('should pend while we wait to GET high-scores', () => {
        state = apiReducer(stateWithScores, actions.getHighScoresPending());
        expect(state).toMatchSnapshot();
    });

    it('should tell us when we successfully GET high-scores', () => {
        state = apiReducer(stateWithScores, actions.getHighScoresSuccess(highScores, 'easyMajor'));
        expect(state).toMatchSnapshot();
    });

    it('should tell us when we fail to GET high-scores', () => {
        state = apiReducer(stateWithScores, actions.getHighScoresFailure(500));
        expect(state).toMatchSnapshot();
    });

    // POST register actions (account creation):
    it('should pend while we wait on account creation (POST)', () => {
        state = apiReducer(initialState, actions.makeUserAccountPending());
        expect(state).toMatchSnapshot();
    });

    it('should tell us when successfully make an account (POST)', () => {
        state = apiReducer(initialState, actions.makeUserAccountSuccess('franklin3', 'auth_token_goes_here'));
        expect(state).toMatchSnapshot();
    });

    it('should tell us when fail to make an account (POST)', () => {
        state = apiReducer(initialState, actions.makeUserAccountFailure(409));
        expect(state).toMatchSnapshot();
    });

    // no CHANGE_PASSWORD* endpoints, but we haven't incorporated that functionality into the client

});
