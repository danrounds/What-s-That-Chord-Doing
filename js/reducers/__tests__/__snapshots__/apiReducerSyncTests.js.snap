// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apiReducer -- the syncronous bits should have a consistent initial state 1`] = `
Object {
  "authToken": null,
  "error": null,
  "highScores": Array [],
  "lastHighScoreAccessed": null,
  "myScores": Object {},
  "name": "null",
  "pending": null,
}
`;

exports[`apiReducer -- the syncronous bits should log off 1`] = `
Object {
  "authToken": null,
  "error": false,
  "highScores": Array [
    Object {
      "name": "ijijijij",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 10,
          "nQuestionNumber": 10,
          "totalClicks": 47,
          "winRatio": 1,
        },
      },
    },
    Object {
      "name": "bobby4",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 124,
          "nQuestionNumber": 129,
          "totalClicks": 12346,
          "winRatio": 0.9612403100775194,
        },
      },
    },
    Object {
      "name": "aseofj_",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 9,
          "nQuestionNumber": 10,
          "totalClicks": 43,
          "winRatio": 0.9,
        },
      },
    },
  ],
  "lastHighScoreAccessed": "hardMajor",
  "myScores": Object {},
  "name": null,
  "pending": false,
}
`;

exports[`apiReducer -- the syncronous bits should pend while we wait for log-in status 1`] = `
Object {
  "authToken": null,
  "error": false,
  "highScores": Array [],
  "lastHighScoreAccessed": null,
  "myScores": Object {},
  "name": "null",
  "pending": true,
}
`;

exports[`apiReducer -- the syncronous bits should pend while we wait on account creation (POST) 1`] = `
Object {
  "authToken": null,
  "error": false,
  "highScores": Array [],
  "lastHighScoreAccessed": null,
  "myScores": Object {},
  "name": "null",
  "pending": true,
}
`;

exports[`apiReducer -- the syncronous bits should pend while we wait to GET high-scores 1`] = `
Object {
  "authToken": undefined,
  "error": false,
  "highScores": Array [
    Object {
      "name": "ijijijij",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 10,
          "nQuestionNumber": 10,
          "totalClicks": 47,
          "winRatio": 1,
        },
      },
    },
    Object {
      "name": "bobby4",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 124,
          "nQuestionNumber": 129,
          "totalClicks": 12346,
          "winRatio": 0.9612403100775194,
        },
      },
    },
    Object {
      "name": "aseofj_",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 9,
          "nQuestionNumber": 10,
          "totalClicks": 43,
          "winRatio": 0.9,
        },
      },
    },
  ],
  "lastHighScoreAccessed": "hardMajor",
  "myScores": Object {
    "name": "joey",
    "scores": Object {
      "allChordsInv": Object {
        "nAnsweredRight": 3,
        "nQuestionNumber": 10,
        "totalClicks": 22,
        "winRatio": 0.3,
      },
      "easyMajor": Object {
        "nAnsweredRight": 4,
        "nQuestionNumber": 10,
        "totalClicks": 12,
        "winRatio": 0.4,
      },
    },
  },
  "name": "joey",
  "pending": true,
}
`;

exports[`apiReducer -- the syncronous bits should pend while we wait to GET my-scores 1`] = `
Object {
  "authToken": undefined,
  "error": false,
  "highScores": Array [
    Object {
      "name": "ijijijij",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 10,
          "nQuestionNumber": 10,
          "totalClicks": 47,
          "winRatio": 1,
        },
      },
    },
    Object {
      "name": "bobby4",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 124,
          "nQuestionNumber": 129,
          "totalClicks": 12346,
          "winRatio": 0.9612403100775194,
        },
      },
    },
    Object {
      "name": "aseofj_",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 9,
          "nQuestionNumber": 10,
          "totalClicks": 43,
          "winRatio": 0.9,
        },
      },
    },
  ],
  "lastHighScoreAccessed": "hardMajor",
  "myScores": Object {
    "name": "joey",
    "scores": Object {
      "allChordsInv": Object {
        "nAnsweredRight": 3,
        "nQuestionNumber": 10,
        "totalClicks": 22,
        "winRatio": 0.3,
      },
      "easyMajor": Object {
        "nAnsweredRight": 4,
        "nQuestionNumber": 10,
        "totalClicks": 12,
        "winRatio": 0.4,
      },
    },
  },
  "name": "joey",
  "pending": true,
}
`;

exports[`apiReducer -- the syncronous bits should tell us when fail to make an account (POST) 1`] = `
Object {
  "authToken": null,
  "error": 409,
  "highScores": Array [],
  "lastHighScoreAccessed": null,
  "myScores": Object {},
  "name": "null",
  "pending": false,
}
`;

exports[`apiReducer -- the syncronous bits should tell us when login-in fails 1`] = `
Object {
  "authToken": null,
  "error": 401,
  "highScores": Array [],
  "lastHighScoreAccessed": null,
  "myScores": Object {},
  "name": null,
  "pending": false,
}
`;

exports[`apiReducer -- the syncronous bits should tell us when login-in succeeds 1`] = `
Object {
  "authToken": "some_string_of_token_or_other2DFS3",
  "error": false,
  "highScores": Array [],
  "lastHighScoreAccessed": null,
  "myScores": Object {},
  "name": "jerry",
  "pending": false,
}
`;

exports[`apiReducer -- the syncronous bits should tell us when successfully make an account (POST) 1`] = `
Object {
  "authToken": "auth_token_goes_here",
  "error": false,
  "highScores": Array [],
  "lastHighScoreAccessed": null,
  "myScores": Object {
    "name": "franklin3",
  },
  "name": "franklin3",
  "pending": false,
}
`;

exports[`apiReducer -- the syncronous bits should tell us when we PUT our updated scores 1`] = `
Object {
  "authToken": undefined,
  "error": false,
  "highScores": Array [
    Object {
      "name": "ijijijij",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 10,
          "nQuestionNumber": 10,
          "totalClicks": 47,
          "winRatio": 1,
        },
      },
    },
    Object {
      "name": "bobby4",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 124,
          "nQuestionNumber": 129,
          "totalClicks": 12346,
          "winRatio": 0.9612403100775194,
        },
      },
    },
    Object {
      "name": "aseofj_",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 9,
          "nQuestionNumber": 10,
          "totalClicks": 43,
          "winRatio": 0.9,
        },
      },
    },
  ],
  "lastHighScoreAccessed": "hardMajor",
  "myScores": Object {
    "name": "joey",
    "scores": Object {
      "allChordsInv": Object {
        "nAnsweredRight": 3,
        "nQuestionNumber": 10,
        "totalClicks": 22,
        "winRatio": 0.3,
      },
      "easyMajor": Object {
        "nAnsweredRight": 4,
        "nQuestionNumber": 10,
        "totalClicks": 12,
        "winRatio": 0.4,
      },
    },
  },
  "name": "joey",
  "pending": false,
}
`;

exports[`apiReducer -- the syncronous bits should tell us when we fail to GET high-scores 1`] = `
Object {
  "authToken": undefined,
  "error": 500,
  "highScores": Array [
    Object {
      "name": "ijijijij",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 10,
          "nQuestionNumber": 10,
          "totalClicks": 47,
          "winRatio": 1,
        },
      },
    },
    Object {
      "name": "bobby4",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 124,
          "nQuestionNumber": 129,
          "totalClicks": 12346,
          "winRatio": 0.9612403100775194,
        },
      },
    },
    Object {
      "name": "aseofj_",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 9,
          "nQuestionNumber": 10,
          "totalClicks": 43,
          "winRatio": 0.9,
        },
      },
    },
  ],
  "lastHighScoreAccessed": "hardMajor",
  "myScores": Object {
    "name": "joey",
    "scores": Object {
      "allChordsInv": Object {
        "nAnsweredRight": 3,
        "nQuestionNumber": 10,
        "totalClicks": 22,
        "winRatio": 0.3,
      },
      "easyMajor": Object {
        "nAnsweredRight": 4,
        "nQuestionNumber": 10,
        "totalClicks": 12,
        "winRatio": 0.4,
      },
    },
  },
  "name": "joey",
  "pending": false,
}
`;

exports[`apiReducer -- the syncronous bits should tell us when we fail to GET my-scores 1`] = `
Object {
  "authToken": undefined,
  "error": "here's an error",
  "highScores": Array [
    Object {
      "name": "ijijijij",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 10,
          "nQuestionNumber": 10,
          "totalClicks": 47,
          "winRatio": 1,
        },
      },
    },
    Object {
      "name": "bobby4",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 124,
          "nQuestionNumber": 129,
          "totalClicks": 12346,
          "winRatio": 0.9612403100775194,
        },
      },
    },
    Object {
      "name": "aseofj_",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 9,
          "nQuestionNumber": 10,
          "totalClicks": 43,
          "winRatio": 0.9,
        },
      },
    },
  ],
  "lastHighScoreAccessed": "hardMajor",
  "myScores": Object {
    "name": "joey",
    "scores": Object {
      "allChordsInv": Object {
        "nAnsweredRight": 3,
        "nQuestionNumber": 10,
        "totalClicks": 22,
        "winRatio": 0.3,
      },
      "easyMajor": Object {
        "nAnsweredRight": 4,
        "nQuestionNumber": 10,
        "totalClicks": 12,
        "winRatio": 0.4,
      },
    },
  },
  "name": "joey",
  "pending": false,
}
`;

exports[`apiReducer -- the syncronous bits should tell us when we fail to PUT our updated scores 1`] = `
Object {
  "authToken": undefined,
  "error": 400,
  "highScores": Array [
    Object {
      "name": "ijijijij",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 10,
          "nQuestionNumber": 10,
          "totalClicks": 47,
          "winRatio": 1,
        },
      },
    },
    Object {
      "name": "bobby4",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 124,
          "nQuestionNumber": 129,
          "totalClicks": 12346,
          "winRatio": 0.9612403100775194,
        },
      },
    },
    Object {
      "name": "aseofj_",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 9,
          "nQuestionNumber": 10,
          "totalClicks": 43,
          "winRatio": 0.9,
        },
      },
    },
  ],
  "lastHighScoreAccessed": "hardMajor",
  "myScores": Object {
    "name": "joey",
    "scores": Object {
      "allChordsInv": Object {
        "nAnsweredRight": 3,
        "nQuestionNumber": 10,
        "totalClicks": 22,
        "winRatio": 0.3,
      },
      "easyMajor": Object {
        "nAnsweredRight": 4,
        "nQuestionNumber": 10,
        "totalClicks": 12,
        "winRatio": 0.4,
      },
    },
  },
  "name": "joey",
  "pending": false,
}
`;

exports[`apiReducer -- the syncronous bits should tell us when we successfully GET high-scores 1`] = `
Object {
  "authToken": undefined,
  "error": false,
  "highScores": Array [
    Object {
      "name": "seofij",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 210,
          "nQuestionNumber": 1000,
          "totalClicks": 4127,
          "winRatio": 1,
        },
      },
    },
    Object {
      "name": "jake",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 1245,
          "nQuestionNumber": 1291,
          "totalClicks": 123466,
          "winRatio": 0.9612403100775194,
        },
      },
    },
    Object {
      "name": "aseos25DSDfj_",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 90,
          "nQuestionNumber": 100,
          "totalClicks": 433,
          "winRatio": 0.9,
        },
      },
    },
  ],
  "lastHighScoreAccessed": "easyMajor",
  "myScores": Object {
    "name": "joey",
    "scores": Object {
      "allChordsInv": Object {
        "nAnsweredRight": 3,
        "nQuestionNumber": 10,
        "totalClicks": 22,
        "winRatio": 0.3,
      },
      "easyMajor": Object {
        "nAnsweredRight": 4,
        "nQuestionNumber": 10,
        "totalClicks": 12,
        "winRatio": 0.4,
      },
    },
  },
  "name": "joey",
  "pending": false,
}
`;

exports[`apiReducer -- the syncronous bits should tell us when we successfully GET my-scores 1`] = `
Object {
  "authToken": undefined,
  "error": false,
  "highScores": Array [
    Object {
      "name": "ijijijij",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 10,
          "nQuestionNumber": 10,
          "totalClicks": 47,
          "winRatio": 1,
        },
      },
    },
    Object {
      "name": "bobby4",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 124,
          "nQuestionNumber": 129,
          "totalClicks": 12346,
          "winRatio": 0.9612403100775194,
        },
      },
    },
    Object {
      "name": "aseofj_",
      "scores": Object {
        "intermediateMinor": Object {
          "nAnsweredRight": 9,
          "nQuestionNumber": 10,
          "totalClicks": 43,
          "winRatio": 0.9,
        },
      },
    },
  ],
  "lastHighScoreAccessed": "hardMajor",
  "myScores": Object {
    "name": "joey",
    "scores": Object {
      "allChordsInv": Object {
        "nAnsweredRight": 13,
        "nQuestionNumber": 20,
        "totalClicks": 33,
        "winRatio": 0.3,
      },
      "easyMajor": Object {
        "nAnsweredRight": 14,
        "nQuestionNumber": 20,
        "totalClicks": 22,
        "winRatio": 0.4,
      },
    },
  },
  "name": "joey",
  "pending": false,
}
`;
